### PR TITLE
Auto-generate workflows in genpkg from dir (SYN-3106)

### PR DESCRIPTION
**Pull request by mikemoritz**

### CODE REVIEW

File: synapse/tools/genpkg.py [hunk 1]
```diff
@@ -44,6 +44,23 @@ def loadOpticFiles(pkgdef, path):
...
 
+def loadOpticWorkflows(pkgdef, path):
+
+    wdefs = pkgdef['optic']['workflows']
+
+    for root, dirs, files in os.walk(path):
+
+        for name in files:
+
+            if name.startswith('.'):  # pragma: no cover
```
Comment (vEpiphyte) [inline synapse/tools/genpkg.py:55]: is there a case where we have `.yaml` files in a workflows dir that starts with a `.` ?

Comment (vEpiphyte) [reply synapse/tools/genpkg.py:55]: we could just skip all files that are not named `.yaml` instead?

Comment (mikemoritz) [reply synapse/tools/genpkg.py]: We could do something simple like `^([a-zA-Z0-9_-]+)\.yaml$` and log a warning for non-matching filenames

Comment (vEpiphyte) [reply synapse/tools/genpkg.py]: I think sounds reasonable for now. It's not overly restrictive from future expansion.

Comment (mikemoritz) [reply synapse/tools/genpkg.py]: updated

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**